.github/ci-scripts/bash/functions-for-build

```diff
@@ -91,6 +117,43 @@ setup_qa_env() {
   deactivate
 }
 
+terraform_apply() {
+  set -e
+  ARTIFACT="${1}"
+  EXTENSION="${2}"
+  requiredInputs="1 2"
+  validateInputs "$requiredInputs" "$@" 
+  aws_configure_devqa_infra
+  ARTIFACT_FILE=$PWD/build/${ARTIFACT}.${EXTENSION}
+  mkdir -p build
+  cd terraform
+  create_tf_backend "tf-state-convoy-dev-deployment" "tf-state-convoy-dev-deployment/terraform.tfstate-md5" "${COMMON_INFRA_ACCESS_KEY}" "${COMMON_INFRA_SECRET_KEY}"
+  terraform init
+  terraform plan
+  terraform_build $PWD ${ARTIFACT_FILE}
+  cd ..
+  ls build
+  set +e
+}
+
+terraformApplyAndPush() {
+  set -e
+  ARTIFACT="$1"
+  EXTENSION="$2"
+  aws_configure_devqa_infra
+  ARTIFACT_FILE=$PWD/build/${ARTIFACT}.${EXTENSION}
+  mkdir -p build
+  cd terraform
+  create_tf_backend "tf-state-convoy-dev-deployment" "tf-state-convoy-dev-deployment/terraform.tfstate-md5" "${COMMON_INFRA_ACCESS_KEY}" "${COMMON_INFRA_SECRET_KEY}"
+  terraform init
+  terraform plan
+  terraform apply -auto-approve
+  terraform_build $PWD ${ARTIFACT_FILE}
```

Comment:
Why do we need to create zip and push?

Reply:
No particular need but just to keep a track, can be removed.